src/types.ts

```diff
@@ -12,8 +14,9 @@ export interface SmartweaveContractSource {
 }
 
 export type SearchBarProps = {
-  buttonAction: () => void;
+  buttonAction: (props: string) => void;
   placeholderText: string;
-  headerText: string;
+  headerElement: JSX.Element;
   footerText: string;
+  availability: string;
```

Comment:
i'm going back and forth on something is needed as a prop to the SearchBar - it should determine this internally and not need it to be provided. what do you think?

Reply:
this is why i suggested it as a global state.

Reply:
it doesn't need to be global - it can be just within `SearchBar` and passed to the `header` component directly

Reply:
so get the records in searchbar?